src/MoBi.Core/Commands/AddBuildingBlocksToModuleCommand.cs

```diff
@@ -0,0 +1,125 @@
+﻿using MoBi.Assets;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Events;
+using MoBi.Core.Helper;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Commands
+{
+   public class AddBuildingBlocksToModuleCommand : MoBiReversibleCommand, ISilentCommand
+   {
+      protected Module _existingModule;
+      protected Module _moduleWithNewBuildingBlocks;
+      public string ExistingModuleId { get; private set; }
+      public string ModuleWithNewBuildingBlocksId { get; private set; }
+
+      public AddBuildingBlocksToModuleCommand(Module existingModule, Module moduleWithNewBuildingBlocks)
+      {
+         ObjectType = new ObjectTypeResolver().TypeFor<Module>();
+         CommandType = AppConstants.Commands.AddCommand;
+         _existingModule = existingModule;
+         _moduleWithNewBuildingBlocks = moduleWithNewBuildingBlocks;
+         ExistingModuleId = existingModule.Id;
+         ModuleWithNewBuildingBlocksId = moduleWithNewBuildingBlocks.Id;
+         Silent = false;
+         Description = AppConstants.Commands.AddBuildingBlocksToModule(_existingModule.Name);
+      }
+
+      //ToDo: this should be refactored to use a generic method - ideally also reflection 
+      //to go through the class members
+      protected override void ExecuteWith(IMoBiContext context)
+      {
+         context.Register(_existingModule);
```

Comment:
?why are you registering n the command/ They arlead exist right?

Reply:
removed